src/shared/sam/debugger/typescriptSamDebug.ts

```diff
@@ -106,3 +109,65 @@ export async function makeTypescriptConfig(config: SamLaunchRequestArgs): Promis
 
     return nodejsLaunchConfig
 }
+
+/**
+ * Recursively searches directory and all nested directories until a TypeScript file is found
+ * @param dir Directory to search
+ * @returns true if at least one TypeScript file exists
+ */
+function hasTypeScriptFilesRecursive(dir:string): boolean {
+    const files = readdirSync(dir)
+
+    files.forEach(function (file){
+        if (file.substr(-3) === '.ts'){
+            return true
+        }
+
+        if(statSync(path.join(dir, file)).isDirectory()){
+            if(hasTypeScriptFilesRecursive(path.join(dir, file))) {
+                return true
+            }
+        }
+    })
+    return false
+}
+
+/**
+ * Returns a function closure with the config parameter 
+ * 
+ * For non-template debug configs (target = code), compile the project
+ * using a temporary default tsconfig.json file.
+ */
+async function compileTypeScriptFunc(config: NodejsDebugConfiguration) {
+    return async function () {
```

Comment:
why return an async function from this async function?

Reply:
Refactored, no longer returns a function.